docs/QueryExamples.rst

```diff
@@ -31,3 +31,103 @@ And it is straightforward, but it won't work.  Because Cypher matches one subgra
       .where_not('ID(other_contact) IN first_level_ids')
 
 And there we have a query which is much more verbose than the original code, but accomplishes the goal in a single query.  Having two queries isn't neccessarily bad, so the code's complexity should be weighed against how both versions perform on real datasets.
+
+
+Example 2: Simple Recommendation Engine
+---------------------------------------
+
+If you are interested in more complex collaborative filter methods check out this `article <https://neo4j.com/blog/collaborative-filtering-creating-teams/>`_
+
+Let's assume you have the following schema:
+
+.. code-block:: cypher
+
+    (:User)-[:follow|:skip]->(:Page)
+
+We want to recommend pages for a user to follow based on their current followed pages.
+Constraints:
+
+- We want to include the source of the recommendation. i.e (we recommend you follow X because you follow Y).
+- We want to exclude pages the user has skipped or already follows.
+- The recommended pages must have a name field
+
+Given our schema, we could write the following Cypher to accomplish this:
+
+.. code-block:: cypher
+
+    match (user:User { id: "1" })
+    match (user)-[:follow]->(followed_page:Page)<-[:follow]-(co_user:User)
+    match (co_user)-[:follow]->(rec_page:Page)
+    WHERE exists(rec_page.name)
+    AND NOT (user)-[:follow|:skip]->(rec_page)
+    with rec_page, count(rec_page) as score, collect(followed_page.name) as source_names
+    ORDER BY score DESC LIMIT {limit}
+    unwind source_names as source_name
+    with rec_page, score, source_name, count(source_name) as contrib
+    with rec_page, score, apoc.coll.sortMaps(collect({name:source_name, contrib:contrib*-1}), 'contrib') as sources
+    return rec_page.name as name, score, extract(source IN sources[0..3] | source.name) as top_sources,
+      size(sources) as sources_count
+    order by score desc
```

Comment:
To be consistent, could you make all of the keywords (`match` / `with` / `unwind` / `return` / `order by` / `as`) capitalized?